### PR TITLE
Create postgres backup scripts and set up cron jobs

### LINKED ISSUE #395: Periodically save a copy of database in case of disaster
Before NewCantus steps up to replace OldCantus, we should set up something to save a copy of the database at regular intervals, just in case disaster strikes and someone deletes all the chants in the database / etc.

### PR DESCRIPTION
**Pull request by jackyyzhang03**
Resolves #395, #870, #740. 

Created database backup scripts to be run outside container through cron. The scripts are currently set up to keep 7 daily backups, 8 weekly backups, and 12 monthly backups.

Runs `populate_next_chant_fields` and `populate_is_last_chant_in_feast` management commands monthly.

Updates certificates every 6 months.

### CODE REVIEW

File: postgres/scripts/monthly.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin.bash
+
+# This script creates a compressed SQL dump of the database in the /var/lib/postgresql/backups/monthly directory.
+# It then deletes any montly backups older than 7 days.
```
Comment (jacobdgm) [inline postgres/scripts/monthly.sh]: this comment is incorrect - it deletes any monthly backups older than a year.

File: cron/postgres/monthly.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin.bash
+
+# This script creates a compressed SQL dump of the database in the /home/ubuntu/backups/postgres/montly directory.
```
Comment (jacobdgm) [inline cron/postgres/monthly.sh]: Here, and elsewhere in this file, "monthly" rather than "montly"

**Review by jacobdgm — CHANGES_REQUESTED** [review]

**Comment by jacobdgm** [comment]
Also, would it make sense to create yearly backups that we keep indefinitely?

I'm also thinking of where certbot things will fit into this: does it make sense to address #870 within this PR, or to merge this PR and create a new one for certbot? i understand this needs to be run a little bit more often than yearly (something like every 9 months?)

**Comment by jackyyzhang03** [comment]
>Also, would it make sense to create yearly backups that we keep indefinitely?

I added another job to keep yearly backups for a decade.

>I'm also thinking of where certbot things will fit into this

Not sure, though if you think it is a good idea, we can do it in this PR along with #740 and maybe #877?

File: cron/cron.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# Note: This is set up to run on the production server. If you want to run it on your local machine, you will need to change the paths.
+
+# min   hour    day     month   weekday command
+0       4       *       *       *       bash /home/ubuntu/code/CantusDB/cron/postgres/daily.sh
+5       4       *       *       0       bash /home/ubuntu/code/CantusDB/cron/postgres/weekly.sh
+10      4       1       *       *       bash /home/ubuntu/code/CantusDB/cron/postgres/monthly.sh
+15      4       1       1       *       bash /home/ubuntu/code/CantusDB/cron/postgres/yearly.sh
+20      4       1       *       *       bash /home/ubuntu/code/CantusDB/cron/management/manage.sh populate_is_last_chant_in_feast
+25      4       1       *       *       bash /home/ubuntu/code/CantusDB/cron/management/manage.sh populate_next_chant_fields
```
Comment (jacobdgm) [inline cron/cron.txt]: Good thought to ensure these don't all run at the same time! Can we run `populate_next_chant_fields` before `populate_is_last_chant_in_feast` (we need up-to-date data on which chants follow each other before we can accurately calculate which chants end feasts)?

**Review by jacobdgm — CHANGES_REQUESTED** [review]
This most recent round of changes looks quite good! I left just one comment with a requested change. Thanks for taking the time to set this up!

File: cron/cron.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+######################
+
+# Note: This is set up to run on the production server. If you want to run it on your local machine, you will need to change the paths.
+
+# min   hour    day     month   weekday command
+0       4       *       *       *       bash /home/ubuntu/code/CantusDB/cron/postgres/daily.sh
+10      4       *       *       0       bash /home/ubuntu/code/CantusDB/cron/postgres/weekly.sh
+20      4       1       *       *       bash /home/ubuntu/code/CantusDB/cron/postgres/monthly.sh
+30      4       1       1       *       bash /home/ubuntu/code/CantusDB/cron/postgres/yearly.sh
+40      4       1       *       *       bash /home/ubuntu/code/CantusDB/cron/management/manage.sh populate_is_last_chant_in_feast; bash /home/ubuntu/code/CantusDB/cron/management/manage.sh populate_next_chant_fields
```
Comment (jacobdgm) [inline cron/cron.txt]: This appears to be back to calculating `is_last_chant_in_feast` before calculating `next_chant`? `is_last_chant_in_feast` depends on `next_chant`.

**Review by jacobdgm — APPROVED** [review]
To me, this looks like it does what we need it to! I'll wait before merging until @kqct has checked this over to verify that the certbot setup is correct.

**Comment by meow10811** [comment]
The certbot setup looks correct, but I'd caution that SSL certificates expire in 90 days -- maybe run the renewal script every 2 months instead of every 6?

**Comment by jacobdgm** [comment]
> maybe run the renewal script every 2 months instead of every 6?

@jackyyzhang03

**Review by kqct — APPROVED** [review]

**Comment by ahankinson** [comment]
Just a note to say that you should run the certbot script much more frequently than that. You can run it every night, or at least once a week. If the certificate is not due for renewal then it will bail and nothing will happen, but if you only run it every two months and some transient error happens (certbot servers are down for maintenance; your servers have a network glitch) then you've lost your window and your certificate will expire.

There's no downside to running it daily or weekly.

**Comment by jacobdgm** [comment]
Thanks, @ahankinson - done! #956

**Pull request merged**